File: Algorithms/C++/02215-Find the Difference of Two Arrays.cpp
class Solution {
public:
    vector<vector<int>> findDifference(vector<int>& nums1, vector<int>& nums2) {
        std::unordered_set<int> set1{nums1.begin(), nums1.end()};
        std::unordered_set<int> set2{nums2.begin(), nums2.end()};

        std::vector<std::vector<int>> result(2);

        // build first result 
        for(const auto& num : set1){
            if(!set2.contains(num)) result[0].push_back(num);
        }

        // build second result 
        for(const auto& num : set2){
            if(!set1.contains(num)) result[1].push_back(num);
        }

        return result;
    }
};

/*
    tc: O(n)
    sc: O(n)

    hash set for both nums1 and nums2 - set1 / set2
*/

File: Algorithms/Python3/02215-Find the Difference of Two Arrays.py
class Solution:
    def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        set1 = set(nums1)
        set2 = set(nums2)
        answer = [] # return this
        result = set()
        for num in nums1:
            if num not in set2:
                result.add(num)
        
        answer.append(list(result))
        result.clear()
        for num in nums2:
            if num not in set1:
                result.add(num)

        answer.append(list(result))
        return answer
    
'''
    tc: O(n)
    sc: O(n)
    use two sets

    to find answer[0]
        - make a set for nums2
        - iterate nums1 to find any nums not in set2
        - place into new set
        
    to find answer[1]
        - make a set for nums1
        - iterate nums2 to find any nums not in set1
        - place into new set

'''